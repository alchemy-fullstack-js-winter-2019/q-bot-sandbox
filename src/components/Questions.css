.qBotTable {
  margin: 1em 0;
  width: 100%;
  overflow: hidden;
  background: #FFF;
  color: black;
  border-radius: 10px;
  box-shadow: 0 8px 6px -6px black;
}

.tableRow {
  border: 1px solid #D9E4E6;
}

.tableRow:hover {
  background-color: #ececff;
}

.tableHeader {
  display: none;
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
  border: 1px solid #FFF;
  background-color: #39499B;
  color: #FFF;
  padding: 1em;
}

.tableHeader:first-child {
  display: table-cell;
  text-align: center;
}

.tableHeader:nth-child(2) {
  display: table-cell;
  text-align: center;
}

.tableHeader:nth-child(3) {
  text-align: center;
}

.tableHeader:nth-child(4) {
  text-align: center;
}

.tableHeader:nth-child(2) span {
  display: none;
}

.tableHeader:nth-child(2):after {
  content: attr(data-th);
}

@media (min-width: 480px) {
  .tableHeader:nth-child(2) span {
    display: block;
  }
  .tableHeader:nth-child(2):after {
    display: none;
  }
}

.tableData {
  display: block;
  word-wrap: break-word;
  max-width: 7em;
}

.tableData:first-child {
  display: table-cell;
  text-align: center;
}

@media (min-width: 480px) {
  .tableData {
    border: 1px solid #D9E4E6;
  }
}

.tableHeader, .tableData {
  text-align: left;
  margin: .5em 1em;
}

@media (min-width: 480px) {
  .tableHeader, .tableData {
    display: table-cell;
    padding: 1em;
  }
}

.taButtonActive {
  float: right;
  background-color: #ABFF00;
  border-radius: 50%;
  width: 2.5em;
  height: 2.5em;
  box-shadow: rgba(0, 0, 0, 0.2) 0 -1px 7px 1px, inset #304701 0 -1px 9px, #89FF00 0 2px 12px;
}

.taButtonActive:hover {
  background-color: rgb(86, 162, 0);
}

.taButton {
  float: right;
  background-color: #F00;
  border-radius: 50%;
  width: 2.5em;
  height: 2.5em;
  box-shadow: rgba(0, 0, 0, 0.2) 0 -1px 7px 1px, inset #441313 0 -1px 9px, rgba(255, 0, 0, 0.5) 0 2px 12px;
}

.taDeleteButton {
  display: -webkit-flex;
  display: flex;
  margin: 0 auto;
  background: transparent;
  border: 1px solid #f00;
  border-radius: 2em;
  color: #f00;
  font-size: 12px;
  height: 2em;
  width: 2em;
  cursor: pointer;
}

.taButton:hover {
  background-color: #A00;
}

body {
  padding: 0 2em;
  font-family: Arial, sans-serif;
  color: #666666;
  background: #f2f2f2;
}

.submit {
  padding:5px 15px; 
  border:2px solid #ccc;
  margin-left: 10px; 
  cursor:pointer;
  border-radius: 5px;
}

input {
  padding:5px; 
  border:2px solid #ccc; 
  border-radius: 5px;
  margin-left: 10px;
}

label {
  margin-left: 10px;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  color: #555;
}

h1 {
  font-family: Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: bold;
  color: #39499B;

}