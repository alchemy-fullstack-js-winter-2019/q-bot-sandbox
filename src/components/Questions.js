import React from 'react';
import { connectFirestore } from './connectFirestore';
import { channelCollection } from '../services/firebase';

 export default function Questions({ channel }) {
  const questionListItems = channel && channel.map(c => {
    const question = c.question.split('> ')[1];
    return (
      <li key={c.id}>
        <p>{c.name}</p>
        <p>{question}</p>
      </li>
    )
  })

   return (
    <ul>
      {questionListItems}
    </ul>
  )
}

 export const ConnectQuestions = connectFirestore(
  channelCollection, 'channel'
)(Questions)
