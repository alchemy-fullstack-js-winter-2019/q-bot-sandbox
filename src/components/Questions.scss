//colors

//table-row-colors
$color_geyser: #d9e4e6;
$white: #fff;
$black: black;
$color_titan_white: #ececff;
//table-header-color & h1 color
$color_sapphire: #39499b;
//claimed-ta-button-colors
$color_lime: #abff00;
$color_clover: #304701;
$color_chartreuse: #89ff00;
$black_20: rgba(0, 0, 0, 0.2);
$color_christi: rgb(86, 162, 0);
//unclaimed-ta-button-colors
$red: #f00;
$color_van_cleef: #441313;
$red_50: rgba(255, 0, 0, 0.5);
$color_bright_red: #a00;
//body-colors
$color_storm_dust: #666666;
$concrete: #f2f2f2;
//input-colors
$color_celeste: #ccc;
$color_fuscous_gray: #555;

//fonts

//imports
@import url('https://fonts.googleapis.com/css?family=Roboto+Mono:400,700');

//header fonts
$font_0: 'Roboto Mono';
$font_1: monospace;
//table font
$font_2: Arial;


// general styling
body {
	padding: 0 2em;
	font-family: $font_2;
	color: $color_storm_dust;
	background: $concrete;
}


input {
	padding: 5px;
	border: 2px solid $color_celeste;
	border-radius: 5px;
	margin-left: 10px;
}

label {
	margin-left: 10px;
	color: $color_fuscous_gray;
}

h1 {
	font-family: $font_0, $font_1;
	font-weight: bold;
	color: $color_sapphire;
}

.qBotTable {
	margin: 1em 0;
	width: 100%;
	overflow: hidden;
	background: $white;
	color: $black;
	border-radius: 10px;
	box-shadow: 0 8px 6px -6px $black;
}

.tableRow {
	border: 1px solid $color_geyser;
	&:hover {
		background-color: $color_titan_white;
	}
}

.tableHeader {
	display: table-cell;
	border-top-left-radius: 10px;
	border-top-right-radius: 10px;
	border: 1px solid $white;
	background-color: $color_sapphire;
	color: $white;
	padding: 1em;
	text-align: left;
	margin: .5em 1em;
	&:first-child {
		display: table-cell;
		text-align: center;
	}
	&:nth-child(2) {
		display: table-cell;
		text-align: center;
		span {
			display: none;
		}
		&:after {
			content: attr(data-th);
		}
	}
	&:nth-child(3) {
		display: table-cell;
		text-align: center;
	}
	&:nth-child(4) {
		display: table-cell;
		text-align: center;
	}
}

.tableData {
	word-wrap: break-word;
	max-width: 7em;
	text-align: left;
	margin: .5em 1em;
	&:first-child {
		display: table-cell;
		text-align: center;
	}
}

.taButtonActive {
	float: right;
	background-color: $color_lime;
	border-radius: 50%;
	width: 2.5em;
	height: 2.5em;
	box-shadow: $black_20 0 -1px 7px 1px, inset $color_clover 0 -1px 9px, $color_chartreuse 0 2px 12px;
	&:hover {
		background-color: $color_christi;
	}
}

.taButton {
	float: right;
	background-color: $red;
	border-radius: 50%;
	width: 2.5em;
	height: 2.5em;
	box-shadow: $black_20 0 -1px 7px 1px, inset $color_van_cleef 0 -1px 9px, $red_50 0 2px 12px;
	&:hover {
		background-color: $color_bright_red;
	}
}

.taDeleteButton {
	display: flex;
	margin: 0 auto;
	background: transparent;
	border: 1px solid $red;
	border-radius: 2em;
	color: $red;
	font-size: 12px;
	height: 2em;
	width: 2em;
	cursor: pointer;
}


.submit {
	padding: 5px 15px;
	border: 2px solid $color_celeste;
	margin-left: 10px;
	cursor: pointer;
	border-radius: 5px;
}

.logout {
	padding: 5px 15px;
	border: 2px solid $color_celeste;
	margin-left: 10px;
	cursor: pointer;
	border-radius: 5px;
	float: right;
}


//mobile styling
@media(max-width: 480px) {
	.tableHeader:nth-child(n + 3) {
		display: none !important;
	}

	.tableData:nth-child(n + 3) {
		display: none !important;
	}
}

@media(min-width: 480px) {
	.tableHeader {
		display: table-cell;
		padding: 1em;
		&:nth-child(2) {
			span {
				display: block;
			}
			&:after {
				display: none;
			}
		}
	}
	.tableData {
		border: 1px solid $color_geyser;
		display: table-cell;
		padding: 1em;
	}
	&:nth-child(2) {
		span {
			display: block;
		}
		&:after {
			display: none;
		}
	}
}
