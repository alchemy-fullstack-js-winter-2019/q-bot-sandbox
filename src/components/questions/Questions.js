import React, { useState } from 'react';
import { useFirestore } from '../connectFirestore';
import { questionCollection } from '../../services/firebase';
import Header from '../layout/Header';
import './Questions.scss';
import Question from './Question';
import FilterForm from './FilterForm';
import SortForm from './SortForm';
import { addClaim } from '../../actions/questions';
import CohortSort from '../CohortSort';

 export default function Questions({ providerData }) {
   const [filterValue, setFilterValue] = useState('')
   const [sortValue, setSortValue] = useState('desc');
   const [cohortSortValue, setCohortSortValue] = useState('')
   const [taName] = useState(providerData[0].displayName);

   const question = useFirestore(questionCollection.orderBy('timestamp', sortValue), [], sortValue, cohortSortValue)
   .filter(c => {
      return (c.question.includes(filterValue.toLowerCase()) || c.question.includes(filterValue.toUpperCase())) && c.channelName.includes(cohortSortValue)
   })
   
   const questionTableItems = question && question.map(doc => {
    return (
      <Question 
        questionObj={doc} 
        key={doc.id} 
        providerData={providerData}
        handleClick={(id) => addClaim(taName, id)}
      />
    )
  })

  const headers = ['Name', 'Question', 'Timestamp', 'TA', 'Solved'];
  const headersList = headers.map((header, i) => {
    return (
      <th className={'tableHeader'} key={i}>
        {header}
      </th>
    )
  })

   return (
    <>
    <Header />
    {question === null && <h1>Loading...</h1>}
    { question && 
      <>
        <FilterForm value={filterValue} onChange={({target}) => setFilterValue(target.value)}/>
        
        <SortForm handleChange={({target}) => setSortValue(target.value)} />

        <CohortSort onChange={({target}) => {setCohortSortValue(target.value)}} />
        
        <h1>TA Queue</h1>
        <table className={'qBotTable'}>
          <thead>
            <tr>
              {headersList}
            </tr>
          </thead>
          <tbody>
            {questionTableItems}
          </tbody>
        </table> 
      </>
    }
  </>
  )
}
