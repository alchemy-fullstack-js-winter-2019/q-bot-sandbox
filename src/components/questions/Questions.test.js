import React from 'react';
import { shallow } from 'enzyme';
import Questions from './Questions';

describe('Questions', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <Questions providerData={'TA Name'}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
