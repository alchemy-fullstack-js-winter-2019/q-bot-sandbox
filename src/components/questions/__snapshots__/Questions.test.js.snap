// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Questions matches a snapshot 1`] = `
<Fragment>
  <WithUser />
  <FilterForm
    onChange={[Function]}
    value=""
  />
  <SortForm
    handleChange={[Function]}
  />
  <CohortSort
    onChange={[Function]}
  />
  <h1>
    TA Queue
  </h1>
  <table
    className="qBotTable"
  >
    <thead>
      <tr>
        <th
          className="tableHeader"
          key="0"
        >
          Name
        </th>
        <th
          className="tableHeader"
          key="1"
        >
          Question
        </th>
        <th
          className="tableHeader"
          key="2"
        >
          Timestamp
        </th>
        <th
          className="tableHeader"
          key="3"
        >
          TA
        </th>
        <th
          className="tableHeader"
          key="4"
        >
          Solved
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</Fragment>
`;
